services/datamanager/builtin/sync_test.go

```diff
@@ -425,29 +431,33 @@ func TestArbitraryFileUpload(t *testing.T) {
 				}
 			}
 
-			// Validate error and URs.
-			// Validate first metadata message.
-			test.That(t, len(fileUploads), test.ShouldEqual, 1)
-			test.That(t, len(urs), test.ShouldBeGreaterThan, 0)
-			actMD := urs[0].GetMetadata()
-			test.That(t, actMD, test.ShouldNotBeNil)
-			test.That(t, actMD.Type, test.ShouldEqual, v1.DataType_DATA_TYPE_FILE)
-			test.That(t, actMD.FileName, test.ShouldEqual, fileName)
-			test.That(t, actMD.FileExtension, test.ShouldEqual, fileExt)
-			test.That(t, actMD.PartId, test.ShouldNotBeBlank)
-
-			// Validate ensuing data messages.
-			dataRequests := urs[1:]
-			var actData []byte
-			for _, d := range dataRequests {
-				actData = append(actData, d.GetFileContents().GetData()...)
-			}
-			test.That(t, actData, test.ShouldResemble, fileContents)
+			if !tc.serviceFail {
+				// Validate first metadata message.
+				test.That(t, len(fileUploads), test.ShouldEqual, 1)
+				test.That(t, len(urs), test.ShouldBeGreaterThan, 0)
+				actMD := urs[0].GetMetadata()
+				test.That(t, actMD, test.ShouldNotBeNil)
+				test.That(t, actMD.Type, test.ShouldEqual, v1.DataType_DATA_TYPE_FILE)
+				test.That(t, actMD.FileName, test.ShouldEqual, fileName)
+				test.That(t, actMD.FileExtension, test.ShouldEqual, fileExt)
+				test.That(t, actMD.PartId, test.ShouldNotBeBlank)
 
-			// Validate file no longer exists.
-			waitUntilNoFiles(additionalPathsDir)
-			test.That(t, len(getAllFileInfos(additionalPathsDir)), test.ShouldEqual, 0)
-			test.That(t, dmsvc.Close(context.Background()), test.ShouldBeNil)
+				// Validate ensuing data messages.
+				dataRequests := urs[1:]
+				var actData []byte
+				for _, d := range dataRequests {
+					actData = append(actData, d.GetFileContents().GetData()...)
+				}
+				test.That(t, actData, test.ShouldResemble, fileContents)
+
+				// Validate file no longer exists.
+				waitUntilNoFiles(additionalPathsDir)
+				test.That(t, len(getAllFileInfos(additionalPathsDir)), test.ShouldEqual, 0)
+				test.That(t, dmsvc.Close(context.Background()), test.ShouldBeNil)
+			} else {
+				// Validate file still exists.
```

Comment:
For here and in the other tests, in this failure case, we should check that len(fileUploads) (or streaming uploads) == 0.

The file could technically still exist since we didn't call `waitUntilNoFiles`.